Source/Logger/trace.cpp

```diff
@@ -280,9 +280,9 @@ STDAPI_(void) HCTraceImplMessage_v(
 
     char message[4096] = {};
 
-    auto result = vsprintf_s(message, format, varArgs);
+    auto result = _vsnprintf_s(message, sizeof(message), _TRUNCATE, format, varArgs);
 
-    if (result < 0)
+    if (result < 0 && strlen(message) != sizeof(message) - 1)
```

Comment:
we should check errno instead

see here https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/vsnprintf-s-vsnprintf-s-vsnprintf-s-l-vsnwprintf-s-vsnwprintf-s-l?view=msvc-170

Reply:
(basically fail if errno is not 0)

```cpp
errno = 0;
<print>

if (errno != 0)
{
    return;
}

Reply:
and also use the template version of the function if at all possible

Reply:
done